sim/core/unit.go

```diff
@@ -417,22 +422,25 @@ func (unit *Unit) MoveTo(moveRange float64, sim *Simulation) {
 		return
 	}
 
-	tickPeriod := 0.5
+	tickPeriod := 0.2
 
 	moveDistance := moveRange - unit.DistanceFromTarget
-	moveSpeed := 2.0
-	timeToMove := math.Abs(moveDistance) / moveSpeed
+	timeToMove := math.Abs(moveDistance) / unit.MoveSpeed
 	moveTicks := timeToMove / tickPeriod
 	moveInterval := moveDistance / float64(moveTicks)
 
 	unit.moveSpell.Cast(sim, unit.CurrentTarget)
 
 	sim.AddPendingAction(NewPeriodicAction(sim, PeriodicActionOptions{
-		Period:          time.Millisecond * 1000 / 7,
-		NumTicks:        int(moveTicks),
+		Period:          time.Millisecond * 200,
+		NumTicks:        int(math.Ceil(moveTicks)),
 		TickImmediately: false,
 
 		OnAction: func(sim *Simulation) {
+			remainingDistance := math.Abs(moveRange - unit.DistanceFromTarget)
+			if(remainingDistance < moveInterval) {
```

Comment:
🔨 In Go you don't have to use parens in a condition like you do in JavaScript so this can just be `if remainingDistance < moveInterval {`